### DISCUSSION TITLE
Plot: Option to lock (sync) x and y axis scale

### DISCUSSION THREAD

**Discussion by ljburtz** [foxglove/community#470] [Q&A] [answered]
plot setting to lock the x and y axis scales (aspect ratio) -> Important for "map" type applications
(when chossing the x axis is "msg path (accumulated)" for example
a square robot trajectory should be a square/not a rectangle! 

~feature 2. fix xmin and xmax as well as ymin and ymax (important for consistent/comparable plots)~ [ @amacneil edit: moved to foxglove/studio#3991 ]

simple additional settings in the 'plot panel settings' would address this
![image](https://user-images.githubusercontent.com/26211495/181675171-ac966812-106c-43e3-bb93-51fa10d5b7a5.png)

**Comment by amacneil** [comment]
@ljburtz let's keep each feature request in a separate ticket. I split the xmin/xmax request into https://github.com/foxglove/studio/issues/3991

**Comment by xavier-guardian** [comment]
Is this feature in development at all? I was wanting it just now!

**Reply by indirectlylit** [reply]
Hi, thank you for the bump on this! I don't see an internal tracking issue so I created FG-10006.

There doesn't seem to be a lot of demand for this, but it certainly makes sense as a need.

**Reply by xavier-guardian** [reply]
yea! thanks for adding it. it's crucial for viewing odometry data like this person mentioned.

**Reply by xavier-guardian** [reply]
@indirectlylit hello! when i use the 'b' macro to scroll both axes, is it holding the ratios constant? It SEEMS like it is, but I wasn't sure? If so, that's a pretty decent stand in for this feature.

I think that since Foxglove was originally designed for looking at long timeseries data, it makes sense that the default scroll would only scroll the horizontal axis. When using plots for XY series tho, that starts to be a little less intuitive, I think. Maybe the default scroll can switch between horizontal mode and "both" mode depending on a flag check in the graph parameters. With the new multiple XY series per plot feature, i am now using Foxglove instead of Plotjuggler to review my XY (odometry) plots, so this is more relevant.

**Reply by indirectlylit** [reply]
Hi @xavier-guardian - yes, the 'b' modifier with scrolling will zoom both axes with locked scales.

That said, we are _very_ close to shipping a feature to lock scales 1:1, which will address this request more directly. When that mode is enabled you will _only_ be able to zoom both axes at once. I'll update this thread when it's ready.

> I think that since Foxglove was originally designed for looking at long timeseries data, it makes sense that the default scroll would only scroll the horizontal axis. When using plots for XY series tho, that starts to be a little less intuitive, I think. Maybe the default scroll can switch between horizontal mode and "both" mode depending on a flag check in the graph parameters

Yes, this is exactly what we've been discussing internally. The eventual behavior will likely have x-axis zoom default for time series data, and locked axis zoom default for XY plots. We're still working through details of how to switch to and communicate the non-default zoom modes.

**Reply by xavier-guardian** [reply]
@indirectlylit yea i've got a good workflow now by zooming until the axes are roughly 1:1 and holding 'b' non-stop. works pretty well. I can finally drop Plotjuggler completely. Thank you.

looking forward to the updates! thanks!

**Comment by indirectlylit** [comment, marked as answer]
This feature has landed! You can now select "1:1" [axis scales](https://docs.foxglove.dev/docs/visualization/panels/plot#settings) for path-based XY plots.

Available now on web, and in the next desktop release.


https://github.com/user-attachments/assets/bf84e1db-180c-4291-83ad-af0cbd4b6b78

**Reply by xavier-guardian** [reply]
works wonderfully! thank you!!
